### ISSUE TITLE
ERR_PACKAGE_PATH_NOT_EXPORTED and import resolution issues

### ISSUE THREAD

**Issue by giannis88** [modelcontextprotocol/typescript-sdk#99]
The SDK's module structure makes it difficult to use in both ESM and CommonJS contexts.

**Issues:**
1. ERR_PACKAGE_PATH_NOT_EXPORTED when importing from root package
2. TypeScript unable to resolve submodule imports (server/index.js, server/stdio.js)
3. Package.json exports field doesn't match internal structure

**Steps to reproduce:**
```typescript
// None of these work:
import { Server } from '@modelcontextprotocol/sdk';
import { Server } from '@modelcontextprotocol/sdk/server';
import { Server } from '@modelcontextprotocol/sdk/dist/server/index.js';
```

**Suggested fixes:**
1. Add proper exports field in package.json
2. Ensure consistent module paths between package.json and build output
3. Add type declarations for submodules
4. Document recommended import patterns

Current workaround is to use require() but this breaks in ESM context.

**Comment by jspahrsummers** [comment]
This is the correct import syntax:
```typescript
import { Server } from '@modelcontextprotocol/sdk/server/index.js';
```

Does that work for you?

**Comment by kushalr392** [comment]
@jspahrsummers No, this specific format gives me the error of ERR_PACKAGE_PATH_NOT_EXPORTED. Running v1.1.0 and node v22.12.0

**Comment by jspahrsummers** [comment]
@kushalr392 That's surprising. Can you please share a full example project that demonstrates the issue?

**Comment by kushalr392** [comment]
@jspahrsummers Thanks for responding, but I resolved the issue. (it was a project where I was testing Kafka stream data with MCP). The client from where I was running the server was running an older version of Node. Updating that fixed the issue immediately.
